File: nice-traffic-light/r/tests/testthat/test-config.R
VALID_UUID <- "11111111-1111-1111-1111-111111111111"

test_that("defaults base URL when environment is unset", {
  config <- load_config(c(VALID_UUID), list(EVAGENE_API_KEY = "evg_test"))

  expect_equal(config$base_url, "https://evagene.net")
  expect_equal(config$api_key, "evg_test")
  expect_equal(config$pedigree_id, VALID_UUID)
  expect_null(config$counselee_id)
})

test_that("honours a custom base URL", {
  config <- load_config(
    c(VALID_UUID),
    list(EVAGENE_API_KEY = "evg_test", EVAGENE_BASE_URL = "https://evagene.example")
  )
  expect_equal(config$base_url, "https://evagene.example")
})

test_that("missing API key raises config_error", {
  expect_error(
    load_config(c(VALID_UUID), list()),
    class = "config_error"
  )
})

test_that("pedigree id must be a UUID", {
  expect_error(
    load_config(c("not-a-uuid"), list(EVAGENE_API_KEY = "evg_test")),
    class = "config_error"
  )
})

test_that("counselee must be a UUID when provided", {
  expect_error(
    load_config(
      c(VALID_UUID, "--counselee", "oops"),
      list(EVAGENE_API_KEY = "evg_test")
    ),
    class = "config_error"
  )
})

File: nice-traffic-light/python/tests/test_config.py
import pytest

from nice_traffic_light.config import DEFAULT_BASE_URL, ConfigError, load_config

_VALID_UUID = "11111111-1111-1111-1111-111111111111"


def test_defaults_base_url_when_env_unset() -> None:
    config = load_config(argv=[_VALID_UUID], env={"EVAGENE_API_KEY": "evg_test"})

    assert config.base_url == DEFAULT_BASE_URL
    assert DEFAULT_BASE_URL == "https://evagene.net"
    assert config.api_key == "evg_test"
    assert config.pedigree_id == _VALID_UUID
    assert config.counselee_id is None


def test_honours_custom_base_url() -> None:
    config = load_config(
        argv=[_VALID_UUID],
        env={"EVAGENE_API_KEY": "evg_test", "EVAGENE_BASE_URL": "https://evagene.example"},
    )

    assert config.base_url == "https://evagene.example"


def test_missing_api_key_raises() -> None:
    with pytest.raises(ConfigError, match="EVAGENE_API_KEY"):
        load_config(argv=[_VALID_UUID], env={})


def test_pedigree_id_must_be_uuid() -> None:
    with pytest.raises(ConfigError, match="pedigree-id"):
        load_config(argv=["not-a-uuid"], env={"EVAGENE_API_KEY": "evg_test"})


def test_counselee_must_be_uuid_when_provided() -> None:
    with pytest.raises(ConfigError, match="counselee"):
        load_config(
            argv=[_VALID_UUID, "--counselee", "oops"],
            env={"EVAGENE_API_KEY": "evg_test"},
        )
